### PR TITLE
Refactor/calendar

### PR DESCRIPTION
**Pull request by MarcinSalamon**
calendar is refactored, models are now pascal cased, deleted many unnecessary fields and lines of code

### CODE REVIEW

**Review by ChineduOzodi — DISMISSED** [review]
Looks good

File: src/app/routes/pages/calendar/calendar.component.ts [hunk 1]
```diff
@@ -103,7 +93,7 @@ export class CalendarComponent implements OnInit, DoCheck {
     // },
```
Comment (YamasakiRhys) [inline src/app/routes/pages/calendar/calendar.component.ts]: What is this commented out curly brace for?

**Review by YamasakiRhys — DISMISSED** [review]
Commented out curly brace in calendar.component.ts on line 103 should be looked at.

File: src/app/routes/pages/calendar/calendar.component.ts (hunk 1 above)
Comment (MarcinSalamon) [reply src/app/routes/pages/calendar/calendar.component.ts]: honestly, i have no clue, wasn't mine

Comment (MarcinSalamon) [reply src/app/routes/pages/calendar/calendar.component.ts]: this isn't fully ready to be merged but I do want to finish it tomorrow at work

Comment (YamasakiRhys) [reply src/app/routes/pages/calendar/calendar.component.ts]: Okay got it.  Thanks

Comment (danielsidhu) [reply src/app/routes/pages/calendar/calendar.component.ts]: maybe just delete it so long as it doesn't affect the functionality of the code

**Review by danielsidhu — DISMISSED** [review]
Looks good. removed a lot of code so I hope it wasn't required for any other components.

**Comment by MarcinSalamon** [comment]
fixed a few performance issues, but calendar population is still slow, there is now a modal that asks if you want to override the event time if there exists an event with the same name, loading text appears when the curriculum list is still loading

File: src/app/routes/routes.module.ts [hunk 2]
```diff
@@ -9,6 +21,7 @@ import { BoomComponent } from '../routes/boom/boom.component';
 import { LoginComponent } from '../routes/login/login.component';
 import { RegisterComponent } from '../routes/register/register.component';
 import { DialogViewComponent } from '../routes/dialog-view/dialog-view.component';
+>>>>>>> 499be7f5a3c51579f8911c996114e466f6ab9acb
```
Comment (danielsidhu) [inline src/app/routes/routes.module.ts]: needs to be removed

File: src/app/routes/routes.module.ts [hunk 3]
```diff
@@ -11,6 +12,16 @@ import { CreateVersionComponent } from './pages/create-version/create-version.co
 import { DialogViewComponent } from './pages/dialog-view/dialog-view.component';
 import { EventDuplicateModalComponent } from './pages/calendar/event-duplicate-modal/event-duplicate-modal.component';
 import { CalendarModalComponent } from './pages/calendar/calendar-modal/calendar-modal.component';
+=======
```
Comment (YamasakiRhys) [inline src/app/routes/routes.module.ts]: Remove please

File: src/app/routes/routes.module.ts [hunk 4]
```diff
@@ -1,6 +1,7 @@
 import { NgModule } from '@angular/core';
 import { CommonModule } from '@angular/common';
 import { Routes, RouterModule } from '@angular/router';
+<<<<<<< HEAD
```
Comment (YamasakiRhys) [inline src/app/routes/routes.module.ts]: Please remove this

File: src/app/routes/routes.module.ts (hunk 2 above)
Comment (MarcinSalamon) [reply src/app/routes/routes.module.ts]: done

File: src/app/routes/routes.module.ts (hunk 4 above)
Comment (MarcinSalamon) [reply src/app/routes/routes.module.ts]: done

**Review by ChineduOzodi — DISMISSED** [review]
Looks good

**Comment by RevatureSteve** [comment]
@MarcinSalamon calendar-event.ts needs an NgDoc providing info and the style guide requirement of your name and batch etc

**Comment by MarcinSalamon** [comment]
please merge this into dev so I don't have to fix conflicts every 30 min today

**Review by ChineduOzodi — APPROVED** [review]
Looks good

**Pull request merged**